### PR TITLE
[TextFields] Convert clear button to MDCButton

### LINKED ISSUE #4845: [TextFields] No touch feedback on x in text input field
This is an internal issue. If you are a Googler, please visit [b/111142484](http://b/111142484) for more details.

---
#### Internal data
- Associated internal bug: [b/111142484](http://b/111142484)

### PR DESCRIPTION
**Pull request by bryanoltman**
Change the "clear" button in `MDCTextInputCommonFundament` from a `UIButton` to an `MDCButton` to add ripple behavior as per the discussion in b/111142484.

Fixes #4845

### CODE REVIEW

File: components/TextFields/src/private/MDCTextInputCommonFundament.m [hunk 1]
```diff
@@ -194,7 +195,19 @@ - (void)commonMDCTextInputCommonFundamentInit {
...
+    MDCButton *clearButton = [[MDCButton alloc] init];
+    clearButton.backgroundColor = UIColor.clearColor;
+    // This ink color was taken from the MDCButton+MaterialTheming behavior, with UIColor.blackColor
+    // taken from the onSurfaceColor value of the MDCColorSchemeDefaultsMaterial201907 color scheme.
+    clearButton.inkColor = [UIColor.blackColor colorWithAlphaComponent:0.12];
+    clearButton.enableRippleBehavior = YES;
+    clearButton.inkStyle = MDCInkStyleUnbounded;
+    clearButton.clipsToBounds = NO;
+    clearButton.contentEdgeInsets = UIEdgeInsetsZero;
+    clearButton.layer.cornerRadius = MDCTextInputClearButtonImageSquareWidthHeight / 2.0f;
```
Comment (jverkoey) [inline components/TextFields/src/private/MDCTextInputCommonFundament.m]: Is this cornerRadius needed?

**Review by jverkoey — CHANGES_REQUESTED** [review]
Please add a snapshot test.

File: components/TextFields/src/private/MDCTextInputCommonFundament.m (hunk 1 above)
Comment (bryanoltman) [reply components/TextFields/src/private/MDCTextInputCommonFundament.m]: Turns out not. Good catch!

**Comment by bryanoltman** [comment]
> Please add a snapshot test.

Done.

File: components/TextFields/src/private/MDCTextInputCommonFundament.m [hunk 2]
```diff
@@ -194,7 +195,18 @@ - (void)commonMDCTextInputCommonFundamentInit {
...
 - (void)setupClearButton {
   if (!_clearButton) {
-    _clearButton = [[UIButton alloc] initWithFrame:CGRectZero];
+    // The following MDCButton configuration creates an MDCButton that mimics a UIButton with a
+    // "clear" icon and adds a ripple effect.
+    MDCButton *clearButton = [[MDCButton alloc] init];
+    clearButton.backgroundColor = UIColor.clearColor;
+    // This ink color was taken from the MDCButton+MaterialTheming behavior, with UIColor.blackColor
+    // taken from the onSurfaceColor value of the MDCColorSchemeDefaultsMaterial201907 color scheme.
+    clearButton.inkColor = [UIColor.blackColor colorWithAlphaComponent:(CGFloat)0.12];
```
Comment (romoore) [inline components/TextFields/src/private/MDCTextInputCommonFundament.m:204]: It looks like there should be a public API for clients to control the ripple color. If their Ripple is a non-gray color throughout their app, this Ripple won't be able to match the other components.  

Would you please add a public API for ripple color in a follow-up PR?

File: components/TextFields/BUILD [hunk 3]
```diff
@@ -51,6 +51,7 @@ mdc_public_objc_library(
     ],
     deps = [
         "//components/AnimationTiming",
+        "//components/Buttons",
         "//components/Elevation",
         "//components/Palettes",
         "//components/Typography",
```
Comment (romoore) [inline components/TextFields/BUILD:57]: Please add a dependency on MaterialButtons and MaterialRipple to the snapshot test targets below.

**Review by romoore — APPROVED** [review]

File: components/TextFields/src/private/MDCTextInputCommonFundament.m (hunk 2 above)
Comment (bryanoltman) [reply components/TextFields/src/private/MDCTextInputCommonFundament.m:204]: Great idea, will do

File: components/TextFields/BUILD (hunk 3 above)
Comment (bryanoltman) [reply components/TextFields/BUILD:57]: Done

**Comment by bryanoltman** [comment]
@romoore Broke the Ripple snapshot test into its own file as per our offline conversation.

File: components/TextFields/src/private/MDCTextInputCommonFundament.m (hunk 2 above)
Comment (bryanoltman) [reply components/TextFields/src/private/MDCTextInputCommonFundament.m:204]: #8794

**Review by romoore — APPROVED** [review]

**Review by jverkoey — APPROVED** [review]

**Pull request merged**